scripts/save-typescript-cache.js

```diff
@@ -0,0 +1,10 @@
+const {exec, gracefulExit} = require('./utilities');
+
+process.on('uncaughtException', gracefulExit);
+process.on('unhandledRejection', gracefulExit);
+
+async function run() {
+  await exec(`tar -cf tsbuildcache/cache.tar packages/*/dist`);
```

Comment:
Do we need to cache the whole `packages/*/dist` directory? Can we get away with just caching the `.tsbuildinfo` files?

Reply:
☝️ also curious about this

Reply:
tsc depends on the presence of dist files.  Example:

```
$ yarn build
$ rm packages/semaphore/dist/src/Semaphore.js
$ yarn build --verbose

#### SNIP other output ####

[11:28:50] Project 'packages/semaphore/tsconfig.json' is out of date because output file 'packages/semaphore/dist/src/Semaphore.js' does not exist

[11:28:50] Building project '/Users/gord/src/github.com/Shopify/quilt/packages/semaphore/tsconfig.json'...
```

Reply:
tsc depends on the presence of dist files.  Example:

```
$ yarn build
$ rm packages/semaphore/dist/src/Semaphore.js
$ yarn build --verbose

#### SNIP other output ####

[11:28:50] Project 'packages/semaphore/tsconfig.json' is out of date because output file 'packages/semaphore/dist/src/Semaphore.js' does not exist

[11:28:50] Building project '/Users/gord/src/github.com/Shopify/quilt/packages/semaphore/tsconfig.json'...
```